src/System.Windows.Extensions/src/System/Media/SoundPlayer.cs

```diff
@@ -502,15 +512,17 @@ private void WorkerThread()
                 _lastLoadException = exception;
             }
 
+            IsLoadCompleted = true;
+            _semaphore.Set();
+
```

Comment:
Is there a specific reason you moved `_semaphore.Set()` up here?

Reply:
Just one of the random issues in the code that happened to be easy to fix.  There's a race condition here where the callback should expect that when it's invoked, the load has completed.  But if IsLoadCompleted and the semaphore are set after the event is queued, then it's possible the event can be invoked before the operation is actually marked as being done.  By swapping the order, that problem goes away.